trpc/queries/goals.ts

```diff
@@ -473,8 +475,8 @@ export const addCalclulatedGoalsFields = (goal: any, activityId: string) => {
     const _isIssuer = goal.activityId === activityId;
     const _lastEstimate = goal.estimate?.length ? goal.estimate[goal.estimate.length - 1].estimate : undefined;
     const _shortId = `${goal.projectId}-${goal.scopeId}`;
-    const _hasAchievementCriteria = goal.goalAchiveCriteria?.length;
-    const _achivedCriteriaWeight = calcAchievedWeight(goal.goalAchiveCriteria ?? []);
+    const _hasAchievementCriteria = !!goal.goalAchiveCriteria?.length;
+    let _achivedCriteriaWeight: number | null = goal.completedCriteriaWeight;
```

Comment:
`const _achivedCriteriaWeight = goal.completedCriteriaWeight == null && goal.goalAchieveCriteria ? calcAchievedWeight(goal.goalAchieveCriteria) : goal.completedCriteriaWeight;`

Reply:
Your decision is a little bit hard to read, however we have same result ;)